filer/models/filemodels.py

```diff
@@ -469,8 +474,8 @@ def restore(self):
             self.folder = filer.models.Folder.objects.get(id=self.folder_id)
 
         old_location, new_location = self.file.name, None
-        destination = self._generate_valid_location_for_restore()
-
+        self._generate_valid_name_for_restore()
+        destination = self.file.field.upload_to(self, self.upload_to_name)
```

Comment:
Well the name '_generate_valid_location_for_restore' is misleading, perhaps instead of 'generate' we can use 'set' as the latter is transparent in having to no return value.